src/watchdog_fsevents.c

```diff
@@ -282,6 +272,60 @@ static void watchdog_pycapsule_destructor(PyObject *ptr)
 }
 
 
+
+/**
+ * Converts a ``CFStringRef`` to a Python string object.
+ *
+ * :param cf_string:
+ *      A ``CFStringRef``.
+ * :returns:
+ *      A Python unicode or utf-8 encoded bytestring object.
+ */
+PyObject * CFString_AsPyUnicode(CFStringRef cf_string_ref)
+{
+    CFIndex cf_string_length;
+    char *c_string_buff = NULL;
+    const char *c_string_ptr;
+    PyObject *py_string;
+
+    c_string_ptr = CFStringGetCStringPtr(cf_string_ref, 0);
+
+    if (G_IS_NULL(c_string_ptr)) {
+        cf_string_length = CFStringGetLength(cf_string_ref);
+        CFStringGetCString(cf_string_ref, c_string_buff, cf_string_length, 0);
+        py_string = PyUnicode_FromString(c_string_buff);
+    } else {
+        py_string = PyUnicode_FromString(c_string_ptr);
+    }
+
+    Py_INCREF(py_string);
```

Comment:
`PyUnicode_FromString` returns a new reference, so there's no need to call `Py_INCREF` here. :-)